src/api/code-challenge/content-types/code-challenge/test-evaluator.ts

```diff
@@ -0,0 +1,279 @@
+
+  overrideConsoleLog((args) => {
+    // Disabled for easier console reading
+    logs.push(args);
+    // @ts-expect-error will fix later
+    console.standardLog("args", ...args);
+  });
+  console.log("\n<challengeLabel>", challengeLabel);
+  console.log("\n\n<<METATEST CODE>>", metaCaseCode);
+  console.log("<formattedCode>", formattedCode);
+  console.log("<formattedCode TYPE>", typeof formattedCode);
+  console.log("<internalTestCode>", internalTestCode, "\n\n");
+  let result;
+
+  // Idea: Figure out programmatically whether metaCode will throw an error.
+  // Tomorrow: make a bool for metaCaseCode errors, if true, allow undefined === userPassed
+
+  // Apr 12
+  // Logic: if visibleMetaError !== null, set metaError === true, if metaError === true, undefined <FINALLY result> is a userError
+  // logic not yet tested on multiple code-challenges
+  let visibleMetaError = null;
+  let existsMetaError = false;
+  try {
+    const metaCheck = eval(formattedCode);
+    console.log("<metaCheck>", metaCheck);
+  } catch (err) {
+    // console.log("<metaCaseCode ERROR>", err);
+    visibleMetaError = err;
+    existsMetaError = true;
+    // if metaError exists, result should be undefined
+    // if no metaError, result should be boolean
+  } finally {
+    console.log("<metaCaseCode ERROR>", visibleMetaError);
+    console.log("<existsMetaError>", existsMetaError);
+  }
+
+  // Delete Later. Will almost always throw error.
+  // try {
+  //   const internalCheck = eval(internalTestCode);
+  //   console.log("<internalCheck>", internalCheck);
+  // } catch (err) {
+  //   console.log("<internalTestCode ERROR>", err);
+  //   // almost every internalTest will be an Error, partial code
+  // }
+  let evalErrorContainer = null;
+  try {
+    // @ts-expect-error will fix later
+    const context = getEvaluationContext(formattedCode, logs);
+
+    // Error: Max Call Stack Exceeded! Why...? Override Console?
+    // console.log("CONTEXT: ", JSON.stringify(context));
+
+    result = evaluateWithContext(
+      `${formattedCode};
+      ${internalTestCode};`,
+      context
+    );
+
+    //
+    // if (
+    //   typeof result === "boolean" &&
+    //   result !== userShouldPass &&
+    //   result === true
+    // ) {
+    //   descriptionMessage = `ERROR: metaTest ${metaTestId} & internalTest ${internalTest.id} are UNEXPECTEDLY ${result}`;
+    //   userPassed = false;
+    //   evaluationError = { message: null, stack: null };
+    // }
+
+    // AKA if result is correct as expected
+
+    // if Error, catch is run...
+  } catch (err) {
+    // if (typeof result !== "boolean") {
+    //   userPassed = false;
+    //   descriptionMessage = `ERROR: FAILED metaTest ${metaTestId} & internalTest ${
+    //     internalTest.id
+    //   } are type: ${(typeof result).toUpperCase()}, should be BOOLEAN`;
+    // }
+    // if Code fails to compile in try block, this can be normal for some EXPECTED negatives
+
+    evalErrorContainer = err;
```

Comment:
why not just call it `evalError`?

Reply:
I renamed the variable to your suggestion.